### ISSUE TITLE
It tests doesn't work in dev mode

### ISSUE THREAD

**Issue by denis-anisimov** [vaadin/flow#5726]
Use this PR to check:
https://github.com/vaadin/flow/pull/5716

Step to reproduce:
- Build Flow.
- go into `flow-tests/test-root-context`.
- run `mvn -f pom-npm.xml jetty:run`
- Open e.g. `http://localhost:8888/view/com.vaadin.flow.uitest.ui.template.EmptyListsView`.

The result is: 404 for `index.js` and `webcomponentsjs/webcomponents-loader.js`.
The expected result: no any 404 for any resource.

I believe the problem is the URLs of the resources and URL of the application (servlet).
Please note that the servlet is mapped to `"/view/*"` (see the view URL above).
And resources URLs on the page are:
- `<script type="text/javascript" src="./../build/webcomponentsjs/webcomponents-loader.js"></script>`
- `<script type="module" defer="" src="./../build/index.js"></script>`.

So the URLs refer to the context root of the application which just doesn't serve them at all.

The fix can be done via adding a servlet which has `"/*"` mapping.
This PR https://github.com/vaadin/flow/pull/5725 contains this workaround and the resources are served correctly (but there are other issues).

**Comment by denis-anisimov** [comment]
Node that tests are passing with a workaround in PR #5725:
https://teamcity.vaadin.com/viewLog.html?buildId=49256&tab=buildResultsDiv&buildTypeId=Flow_FlowPullRequestsExtendedWithNpm

**Comment by bogdanudrescu** [comment]
`index.js` is served as a static file by jetty

**Comment by denis-anisimov** [comment]
>`index.js` is served as a static file by jetty from `src/main/webapp/build`, it doesn't reach `VaadinServlet`

Is it ?
I see that `build` directory goes into the build project folder (inside `target`) which is not set as web resource folder in places where NPM mode works.

Also I don't see how things change if I add a servlet which is mapped to the `"/*"`. 
If `index.js`  should be in web resources folder then it's not there anyway. But with the servlet mapped to `"/*"` it works.

**Comment by bogdanudrescu** [comment]
By debugging it, in `VaadinServlet` there is no invocation when `index.js` is requested by browser.

Then simply placing a file named `index.js` in `flow⁩/flow-tests⁩/⁨test-root-context⁩/⁨src⁩/main⁩/⁨webapp⁩/build` it'll be served by jetty. So I think this is jetty configuration issue, but I'm not sure.

**Comment by denis-anisimov** [comment]
#5740

I think this can be closed as a duplicate.

This is a bug in Flow even if it's a jetty configuration issue.
Because:
- as a user I don't know anything about `index.js`  and `webcomponents-loader.js`.
- those files are generated either in production mode or somehow by dev mode. Which means I don't create them and don't have to know anything about them. As a result I don't care about them.
- since those files are not controlled by me everything should work out of the box if I map a servlet to non root URL. But it doesn't. And I have no idea why.
- we know something about those files (but even I don't know how those files are served in reality: are they static or served by a servlet) but user has no any idea about them. Project just doesn't work.
- there is no any documentation or any pointers what's wrong.

So as a result:
- this is a serious issue (especially if even we don't really know how those files are served in the result)
- if those files are static we resources then it should be documented somewhere and everything explained in the details (what those files are, how to make them available and so on).
- I would say that it's too confusing when I need to make additional project configuration because of files which I don't even control/care (`index.js`  and `webcomponents-loader.js`). So I consider this is as a bug: I expected that everything works out of the box regardless of servlet mapping. And it needs to be fixed somehow.
- if we can't fix this properly with current way of working (if files are static web resources) then I would suggest to serve them via a servlet on the fly instead to avoid any confusion.

**Comment by bogdanudrescu** [comment]
Closing as duplicate of https://github.com/vaadin/flow/issues/5740

**Comment by denis-anisimov** [comment]
No, please don't close this until it's verified.

It's true that #5740 is the main reason.
But this issue technically is not exactly the same as #5740.
So once #5740 is fixed this needs to be verified and closed only after that.
We may not be sure that there are no other issues with our IT test apart #5740.

**Comment by denis-anisimov** [comment]
Blocked by #5740

**Comment by denis-anisimov** [comment]
The ticket itself is fixed.

But I would like to enable test-root-context npm dev mode tests as a part of it to verify that it really works now and there will be no regressions.
